Generics.playground/Contents.swift

```diff
@@ -1,3 +1,55 @@
 import UIKit
 
-var str = "Hello, playground"
+struct CountedSet <Element: Hashable> {
+    
+    private(set) var dictionary = [Element: Int]()
+    
+    var count: Int? {
+        if dictionary.isEmpty {
+            return nil
+        } else {
+            return dictionary.count
+        }
+    }
+    
+    mutating func insert(element: Element) {
+        dictionary[element] = (dictionary[element] ?? 0) + 1
+    }
+    
+    mutating func remove(element: Element) {
+        let count = (dictionary[element] ?? 0)
+        if count > 0 {
+            dictionary[element] = count - 1
+        }
+    }
```

Comment:
Maybe we should remove the element if it ever gets to 0? This may same memory in the long run.